libraries/chain/block_header_state.cpp

```diff
@@ -188,9 +189,13 @@ void finish_next(const block_header_state& prev,
          while (it != prev.finalizer_policies.end() && it->first <= lib) {
             const finalizer_policy_tracker& tracker = it->second;
             if (tracker.state == finalizer_policy_tracker::state_t::pending) {
-               // new finalizer_policy becones active
+               // new finalizer_policy becomes active
                // -----------------------------------
                next_header_state.active_finalizer_policy.reset(new finalizer_policy(*tracker.policy));
+               // new finalizer policy generation
+               parent_block.finalizer_policy_generation = next_header_state.active_finalizer_policy->generation;
```

Comment:
Why do we update the `finalizer_policy_generation` in `parent_block`?

Reply:
Because we use it on the next line to calculate the next core.

Reply:
Calculating the core again is less than ideal, but seemed like the best option. Alternatively we could modify the core directly to update the `finalizer_policy_generation`, but re-calculating the next core seemed better than making the core mutable.

Reply:
I don't have a problem with caIculating the core again.
I don't see where `finalizer_policy_generation` is used in `finality_core::next()`. I must be missing something.

Reply:
https://github.com/AntelopeIO/spring/blob/885a574013ac5617a69c8ce44fa11cda4e18ebe4/libraries/chain/finality/finality_core.cpp#L359-L359

Reply:
Thanks!
But that was not the active policy for **that** block which refers to the parent block!

Reply:
You are correct, this is not needed.